tests/test_octettruss.py

```diff
@@ -1,78 +1,107 @@
-import microgen
+from microgen import Rve, Phase, Cylinder, periodic, fuseShapes, cutPhases, meshPeriodic, is_periodic
 import cadquery as cq
+import pyvista as pv
 import numpy as np
-import os
-
+from typing import Union
 import pytest
+from pathlib import Path
+
+@pytest.fixture(scope='function')
+def rve() -> Rve:
+    return Rve(dim_x=1, dim_y=1, dim_z=1, center=(0.5, 0.5, 0.5))
+
+@pytest.fixture(scope="function")
+def tmp_dir(tmp_path_factory: pytest.TempPathFactory) -> Path:
+    tmp_dir_name = "test_tmp_dir"
+    return tmp_path_factory.mktemp(tmp_dir_name)
 
+@pytest.fixture(scope="function")
+def tmp_output_compound_filename(tmp_dir: Path) -> str:
+    return (tmp_dir / "compound.step").as_posix()
+
+@pytest.fixture(scope="function")
+def tmp_output_vtk_filename(tmp_dir: Path) -> str:
+    return (tmp_dir / "octettruss.vtk").as_posix()
 
 @pytest.mark.filterwarnings("ignore:Object intersecting")
-def test_octettruss():
-
-    # fichier
-    NPhases_file = "examples/Lattices/octetTruss/test_octet.dat"
-
-    dt = np.dtype(
-        [
-            ("number", int),
-            ("shape", np.str_, 10),
-            ("xc", np.float64),
-            ("yc", np.float64),
-            ("zc", np.float64),
-            ("psi", np.float64),
-            ("theta", np.float64),
-            ("phi", np.float64),
-            ("height", np.float64),
-            ("radius", np.float64),
-        ]
-    )
-    # précision du type des données
-    number, shape, xc, yc, zc, psi, theta, phi, height, radius, = np.loadtxt(
-        NPhases_file,
-        dtype=dt,
-        usecols=(0, 1, 2, 3, 4, 5, 6, 7, 8, 9),
-        skiprows=1,
-        unpack=True,
-        ndmin=1,
-    )
+def _generate_cqcompound_octettruss(
+    rve: Rve
+):
+
+    #data to generate octet truss cylinders
+    xc = np.array([0.5,0.5,0.5,0.5,0.0,0.0,0.5,0.5,0.5,0.5,0.0,0.0])
+    yc = np.array([0.5,0.5,0.0,0.0,0.5,0.5,0.5,0.5,0.0,0.0,0.5,0.5])
+    zc = np.array([0.0,0.0,0.5,0.5,0.5,0.5,0.0,0.0,0.5,0.5,0.5,0.5])
+    psi = np.array([45.,-45.,0.,0.,90.,90.,0.,0.,90.,90.,45.,-45.])
+    theta = np.array([0.,0.,-90.,-90.,-90.,-90.,90.,90.,90.,90.,0.,0.])
+    phi = np.array([0.,0.,45.,-45.,45.,-45.,-45.,45.,-45.,45.,0.,0.])
+    height = np.full_like(xc, 2.0)
+    radius = np.full_like(xc, 0.05)
 
-    revel = microgen.Rve(dim_x=1, dim_y=1, dim_z=1, center=(0.5, 0.5, 0.5))
     listPhases = []  # type: list[microgen.Phase]

```

Comment:
Maybe use numpy.typing syntax instead of hinting types in comments:

```python
listPhases : list[Phase] = []
listPeriodicPhases : list[Phase] = []
```

Reply:
Done